### ISSUE TITLE
Rendered Doc live url is 'http://' when served behind load balancer

### ISSUE THREAD

**Issue by autodidacticon** [miguelgrinberg/APIFairy#69]
The `servers` property in the docs seems to be generated from the request [here](https://github.com/miguelgrinberg/APIFairy/blob/main/src/apifairy/core.py#L125). When the docs are served behind a load balancer wherein SSL is terminated, the live url is rendered as `http://` which may not be supported depending on the deployment.

Options seem to be:

* deploy docs separately
* fork project and rewrite scheme from `http://` to `https://` based on some config, etc

This may be a feature request but I wanted to verify my reasoning first.

Is there anything missing or incorrect here?

**Comment by miguelgrinberg** [comment]
This is not a bug, as far as I can see. APIFairy uses the server URL reported by Flask, so to me that looks like you haven't properly configured Flask to generate correct URLs for your site. Don't URLs generated with `url_for()` have the same issue, for example?

One option that you have is to set up the [ProxyFix](https://flask.palletsprojects.com/en/2.2.x/deploying/proxy_fix/) middleware.

**Comment by autodidacticon** [comment]
TLDR: ProxyFix seems like the right answer.

Further questions to see what I'm possibly doing wrong:

>This is not a bug, as far as I can see.

Certainly. This issue was meant to be a question.

> APIFairy uses the server URL reported by Flask, so to me that looks like you haven't properly configured Flask to generate correct URLs for your site.

APIFairy uses the url from the request. In this scenario the request's scheme will be 'http://'. What Flask config should be set to alter that behavior? 

> Don't URLs generated with url_for() have the same issue, for example?

`url_for` appears to allow for the scheme to be set per route but how would that affect the project root ("live url") in the markup?

**Comment by miguelgrinberg** [comment]
> APIFairy uses the url from the request. In this scenario the request's scheme will be 'http://'. What Flask config should be set to alter that behavior?

From Flask's point of view the URLs are http://, because they come from your proxy server without SSL. So Flask's request is correct. The ProxyFix middleware reconfigures the Flask request so that the details of the real client are used instead of nginx.

> url_for appears to allow for the scheme to be set per route but how would that affect the project root ("live url") in the markup?

You can always manually change what url_for generates, but I was just trying to make the point that this function was also going to give you the wrong results unless you add ProxyFix or make a manual override.

**Comment by autodidacticon** [comment]
> From Flask's point of view the URLs are http://, because they come from your proxy server without SSL. So Flask's request is correct.

This was my original point. Yet you indicated something is incorrect here:

> APIFairy uses the server URL reported by Flask, so to me that looks like you haven't properly configured Flask to generate correct URLs for your site.

Is this statement still accurate?
